### ISSUE TITLE
It will only talk once?

### ISSUE THREAD

**Issue by BrainSlugs83** [voxell-tech/UnitySpeechEngine#3]
**Describe the bug**
Calling `textToSpeech.Speak("Hello World!");` only works the first time.
After that, if you want it to talk again, you have to stop and restart playmode in the unity editor.

**To Reproduce**
1. Call `textToSpeech.Speak("Hello World!");` once per second.
2. Notice that it speak the first time, but not on any subsequent calls.

**Expected behavior**
It should speak every time you call the `Speak` function.

**Desktop (please complete the following information):**
 - OS: Windows 10

**Comment by maxg0** [comment]
I used a workaround for this by turning each sentence into an object of its own that gets discarded afterwards. Your exact implementation might differ but the core concept might be the same. I create a pool of speakers so that they can instantiate in peace (as the mappings are read in the `Start()` method of TTSProcessor.cs and I was getting errors when trying to run it right away). It's probably a less-than-optimal solution but it works.

**Comment by nixonyh** [comment]
hey @maxg0 @BrainSlugs83 is this still an issue currently now that I have fixed this #2

**Comment by maxg0** [comment]
@nixon-voxell Yes, the issue still persists. I'm getting no error messages either.

**Comment by nixonyh** [comment]
Ok, I'll take a look on this issue.

**Comment by benhamlett** [comment]
Heya, any potential fixx for this yet?

**Comment by Crazyzooka** [comment]
Looks like it could be an issue with tensorflow library? Tracing through callstack just leads to TfLiteInterpreterGetOutputTensor returning a bunch of 0s after the first call for some reason. Or maybe it could be the fastspeech model?

**Comment by Crazyzooka** [comment]
The issue was located here:
https://github.com/asus4/tf-lite-unity-sample/issues/260#issuecomment-1404928610

From my limited understanding of C#, UnityTTS is passing an Array that gets deallocated at the end of the function when the Interpreter GCHandle still contains a shallow copy of the array in the dictionary. When calling it a second time, the same tensor index is being used and assumes that the array is still active in that tensor index, when there is nothing to use.

![image](https://user-images.githubusercontent.com/50309931/217558525-baf2738d-131e-45b8-9161-935cdb63ae2e.png)
![image](https://user-images.githubusercontent.com/50309931/217558598-ef1622ff-4f07-4773-8ccc-d24657afa659.png)

This is my workaround for now until I have more time to make a more elegant fix. This is not ideal for performance but at least makes the voice trigger every call.

**Comment by nixonyh** [comment]
i see, thanks for the info!

**Comment by BrainSlugs83** [comment]
> This is my workaround for now until I have more time to make a more elegant fix. This is not ideal for performance but at least makes the voice trigger every call.

Seems like not just a performance issue, but a resource leak as well. -- You're calling `GCHandle.Alloc()`, but you're never calling `Free()` on that GCHandle.

Reference: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.gchandle.alloc?view=net-8.0
